k8srm-prototype/pkg/api/capacity_types.go

```diff
@@ -53,6 +61,58 @@ type DevicePoolStatus struct {
 	AvailableDevices int `json:"availableDevices,omitempty"`
 }
 
+// Device is used to track individual devices in a pool.
+type Device struct {
+	// Name is a driver-specific identifier for the device.
+	// +required
+	Name string `json:"name"`
+
+	// Attributes contain additional metadata that can be used in
+	// constraints. If an attribute name overlaps with the pool attribute,
+	// the device attribute takes precedence.
+	//
+	// +optional
+	Attributes []Attribute `json:"attributes,omitempty"`
+
+	// Requests contains the pooled resources that are consumed when
+	// this device is allocated.
+	//
+	// +optional
+	Requests map[string]resource.Quantity `json:"requests,omitempty"`
+
+	// Resources allows the definition of per-device resources that can
+	// be allocated in a manner similar to standard Kubernetes resources.
+	// NOTE: We may not need to implement this right away.
+	//
+	// +optional
+	Resources []ResourceCapacity `json:"resources,omitempty"`
+}
+
+type ResourceCapacity struct {
+	// Name is the resource name/type.
+	Name string `json:"name"`
+
+	// Capacity is the total capacity of the named resource.
+	// +required
+	Capacity resource.Quantity `json:"capacity"`
+
+	// BlockSize is the increments in which capacity is consumed. For
+	// example, if you can only allocate memory in 4k pages, then the
+	// block size should be "4Ki". Default is 1.
+	//
+	// If the resource is consumable in a fractional way, then the
+	// default of 1 should not be used; instead this should be a fractional
+	// amount corresponding the increment size. We may also need a minimum
+	// value, if the minimum is larger than the block size (as is the case
+	// for standard Kubernetes CPU resources).
+	//
+	// +optional
+	BlockSize *resource.Quantity `json:"blockSize,omitempty"`
```

Comment:
I don't have an immediate use for this, but I can see how it might be useful for certain resources.

Reply:
This is a candidate for YAGNI(yet) -- if we don't have a concrete need yet, we can always add it later.